src/lib.rs

```diff
@@ -239,16 +239,30 @@ mod contract {
 
         Engine::check_nonce(&sender, &signed_transaction.transaction.nonce).sdk_unwrap();
 
+        // Check intrinsic gas is covered by transaction gas limit
+        match signed_transaction
+            .transaction
+            .intrinsic_gas(&crate::engine::CONFIG)
+        {
+            None => sdk::panic_utf8(b"ERR_GAS_OVERFLOW"),
+            Some(intrinsic_gas) => {
+                if signed_transaction.transaction.gas < intrinsic_gas.into() {
+                    sdk::panic_utf8(b"ERR_INTRINSIC_GAS")
+                }
+            }
+        }
+
         // Figure out what kind of a transaction this is, and execute it:
         let mut engine = Engine::new_with_state(state, sender);
         let value = signed_transaction.transaction.value;
+        let gas_limit = u256_to_u64(signed_transaction.transaction.gas);
```

Comment:
I'm not sure if this is correct behavior or not. Presently it is taking the gas limit mod `u64::MAX + 1` to make it fit into `u64` (since that is the type expected by SputnikVM). But should I panic if the gas limit is larger than `u64::MAX` instead of silently ignoring the larger number? Or maybe I should truncate to `u64::MAX` in the case it is larger? Is there an Ethereum spec for this?

Reply:
Gas limits are `uint64` throughout the Geth code base, so there's no way it can handle a larger value. However, I'm unsure what Geth does if receiving a transaction with a larger value. I'll endeavor to find out and come back to you.

Reply:
Specifying the same max gas limit with another endpoint implementation:

`{'code': -32000, 'message': 'exceeds block gas limit'}`

Reply:
Looks like Geth specifies the error code `ErrGasUintOverflow` (`gas uint64 overflow`) for situations where the gas would overflow:

https://github.com/ethereum/go-ethereum/blob/master/core/error.go#L64

We probably ought to define a corresponding error code for this, so that we can return the corresponding error message in the relayer, for full Geth compatibility.

Reply:
Made to panic if gas limit is larger than `u64::MAX` in https://github.com/aurora-is-near/aurora-engine/pull/123/commits/57934d44cf42e8e2ecbbd4d32508c83175441d65